source/kernel/src/registry/mod.rs

```diff
@@ -299,37 +363,47 @@ impl Registry {
     ///
     /// Driver services registered with [Registry::register_konly] cannot be retrieved via
     /// a call to [Registry::get_userspace].
+    #[tracing::instrument(
+        name = "Registry::get_userspace",
+        level = "debug",
+        skip(self),
+        fields(uuid = ?RD::UUID),
+    )]
     pub fn get_userspace<RD>(&mut self) -> Option<UserspaceHandle>
     where
         RD: RegisteredDriver,
         RD::Request: Serialize + DeserializeOwned,
         RD::Response: Serialize + DeserializeOwned,
     {
         let item = self.items.iter().find(|i| &i.key == &RD::UUID)?;
+        let client_id = self.counter;
+        info!(uuid = ?RD::UUID, service_id = item.value.service_id, client_id = self.counter, "Got KernelHandle from Registry");
+        self.counter = self.counter.wrapping_add(1);
         Some(UserspaceHandle {
             req_producer_leaked: item.value.req_prod.clone(),
             req_deser: item.value.req_deser?,
+            service_id: item.value.service_id,
+            client_id,
         })
     }
 }
 
 // UserRequest
 
-// UserResponse
-
-impl<U: MaxSize, E: MaxSize> MaxSize for UserResponse<U, E> {
-    const POSTCARD_MAX_SIZE: usize = {
-        <[u8; 16] as MaxSize>::POSTCARD_MAX_SIZE
-            + <u32 as MaxSize>::POSTCARD_MAX_SIZE
-            + <Result<U, E> as MaxSize>::POSTCARD_MAX_SIZE
-    };
-}
-
 // Envelope
 
 impl<P> Envelope<P> {
-    pub fn new(body: P) -> Self {
-        Envelope { body }
+    /// Create a response Envelope from a given request Envelope.
+    ///
+    /// Maintains the same Service ID and Client ID, and increments the
+    /// request ID by one.
```

Comment:
i don't get this behavior...why does a response to a request increment the request ID? shouldn't the response habe the same req ID, so that we know what request it's a response to? or am i missing something?

Reply:
So, my GOAL was to have a predictable but different ID for replies, so they could be distinguished in logs as coming from a different sender. Right now I have client/service id, which doesn't flip on the "response". Essentially this was intended as a (hacky) way to use the lsb as a "request/response" flag.

It might have been poorly executed.

Reply:
Ah, okay, that makes sense, i think. But I would kind of expect any logs mentioning a response ID would also indicate that it's a response through some other mechanism?